### PR TITLE
kvrpc: update errors

### PR DESCRIPTION
**Pull request by disksing**
@ngaut @siddontang @morefreeze

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -24,23 +24,10 @@ message LockInfo {
...
-        Other       = 9;
-    }
-    optional Type type                          = 1 [default = Other];
-    optional string msg                         = 2;
-    // If and only if Type == Locked
-    optional LockInfo lock_info                 = 3;
-    optional errorpb.NotLeader leader_info      = 4;
+message KeyError {
+    optional LockInfo locked    = 1; // Client should backoff or cleanup the lock then retry.
+    optional string   retryable = 2; // Client may restart the txn. e.g write conflict.
```
Comment (siddontang) [inline proto/kvrpcpb.proto]: why string?

Comment (disksing) [inline proto/kvrpcpb.proto]: @siddontang For simplifying handle logic in TiClient.
TiClient does not really care about what causes the error, it just logs the error message then return a retryable error to TiDB.

**Comment by siddontang** [comment]
LGTM

**Comment by morefreeze** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM.

**Pull request merged**